### DISCUSSION TITLE
[Changelog] Group changes by scope when conventional-commits is applied

### DISCUSSION THREAD

**Discussion by aalmiray** [jreleaser/jreleaser#1188] [Ideas]
Had a quick look at the changelog generated for `v1.4.0` and could fail to notice there's a lot of changes per scope. At the moment all changes are listed in chronological order inside each category, like so

```
🚀 Features

 * 29e24c9 gradle: Use BuildServices for displaying the banner
 * 5158b09 core: Support FLAT_BINARY distribution type, closes #1115
 * d46c04e Add init & json-schema commands to Ant, Maven, and Gradle., closes #1123
 * 0e3bc7d changelog: Skip username resolution when running in dry-run mode., closes #1116
 * 84d9ae2 core: Support Zstd compression in archive assemblers. Resolves #1106, closes #1106
 * ddc4d35 changelog: Reference closed issues. Resolves #806, closes #806
 * 5569a22 changelog: Handle BREAKING CHANGE from footer. Resolves #809, closes #809
 * a0ea282 core: Add Java properties to output.properties. Resolves #1085, closes #1085
 * 2cac1a2 core: Display Java version in version banner. Resolves #1084, closes #1084
```

Perhaps it would be better to group them by scope, such as:

```
🚀 Features

 **gradle**
  * 29e24c9 Use BuildServices for displaying the banner

 **core**
   * 5158b09 Support FLAT_BINARY distribution type, closes #1115
   * 84d9ae2 Support Zstd compression in archive assemblers. Resolves #1106, closes #1106
   * a0ea282 Add Java properties to output.properties. Resolves #1085, closes #1085
   * 2cac1a2 Display Java version in version banner. Resolves #1084, closes #1084

 **changelog**
   * 0e3bc7d Skip username resolution when running in dry-run mode., closes #1116
   * ddc4d35 Reference closed issues. Resolves #806, closes #806
   * 5569a22 Handle BREAKING CHANGE from footer. Resolves #809, closes #809

 * d46c04e Add init & json-schema commands to Ant, Maven, and Gradle., closes #1123
```

Grouping would have to be activated explicitly using a property. Presets are not real DSL objects thus there's no direct place to add such property. What could be done is follow the same principle as other DSL obejcts and add `extraProperties` to `changelog`. In this way one could define a key `groupedByScope` (or similar) and have the CC parser honor that setting. As an additional benefit any extra properties defined at the `changelog` may also participate in evaluation of name templates, prefixed with `changelog` of course.

@gotson  WDYT?

**Comment by gotson** [comment]
I like the idea!



> have the CC parser honor that setting

the `ConventionalCommit` objects already have the parsed scope stored, it's more a matter of output if i remember correctly.
